app/utils/log-context.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const { format } = require('winston')
+const {
+  CORRELATION_ID,
+  USER_EXTERNAL_ID
+} = require('@govuk-pay/pay-js-commons').logging.keys
+
+const { AsyncLocalStorage } = require('async_hooks')
+const { CORRELATION_HEADER } = require('../../config')
+
+const asyncLocalStorage = new AsyncLocalStorage()
+
+function logContextMiddleware (req, res, next) {
+  asyncLocalStorage.run({}, () => {
+    asyncLocalStorage.getStore()[CORRELATION_ID] = req.headers[CORRELATION_HEADER]
+    next()
+  })
+}
+
+const supplementSharedLoggingFields = format((info) => {
+  if (!asyncLocalStorage.getStore()) {
+    return info
+  }
+
+  return Object.assign(info, asyncLocalStorage.getStore())
+})
+
+function addUserLoggingFields (user) {
```

Comment:
The pattern before this felt a bit too specific for the logging context utility but it feels like this finds the right balance. For common entities extracting and including info we think is relevant in one place feels quite clean. 

We could potentially do this for services and accounts too for routes that pertain to those. 

Some ideas 

```js 
// service external id
// go live status is "LIVE" yet or not? - unsure if that's useful
function addServiceLoggingFields(service) {}

// live or test
// account id
// account external id?
// payment provider
function addAccountLogginFields(account) {}
```